src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.UserRegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    public static final int MIN_LENGTH_PASSWORD_PARAMETER = 6;
+    public static final int MIN_LENGTH_LOGIN_PARAMETER = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserExist(user);
+        checkUserLoginLength(user);
+        checkUserPassLength(user);
+        checkUserAge(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUserExist(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
You should also check if the user login is not null before calling `storageDao.get(user.getLogin())`. If the login is null, it will throw a NullPointerException. For example, you can throw an exception if the login is null or if a user with such login already exists.